controllers/webhook_test.go

```diff
@@ -0,0 +1,213 @@
+//go:build test_all || test_unit
+
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"os"
+	"testing"
+
+	dspav1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1"
+	"github.com/opendatahub-io/data-science-pipelines-operator/controllers/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	testDSPONamespace = "testDSPONamespace"
+	testWebhookName   = "ds-pipelines-webhook"
+)
+
+func TestMain(m *testing.M) {
+	os.Setenv("DSPO_NAMESPACE", testDSPONamespace)
+	code := m.Run()
+	os.Unsetenv("DSPO_NAMESPACE")
+	os.Exit(code)
+}
+
+func TestWebhookLifecycle(t *testing.T) {
```

Comment:
Does this work instead?
```suggestion
func TestWebhookLifecycle(t *testing.T) {
	t.Setenv("DSPO_NAMESPACE", testDSPONamespace)
```